modules/openapi-generator/src/main/resources/typescript-axios/api.mustache

```diff
@@ -172,12 +172,14 @@ export const {{classname}}AxiosParamCreator = function (configuration?: Configur
     {{#headerParams}}
             {{#isListContainer}}
             if ({{paramName}}) {
-                localVarHeaderParameter['{{baseName}}'] = {{paramName}}.join(COLLECTION_FORMATS["{{collectionFormat}}"]);
+                let mapped = {{paramName}}.map(value => (<any>"{{{dataType}}}" !== "Array<string>") ? JSON.stringify(value || {}) : (value || ""))
+                localVarHeaderParameter['{{baseName}}'] = mapped.join(COLLECTION_FORMATS["{{collectionFormat}}"]);
             }
             {{/isListContainer}}
             {{^isListContainer}}
             if ({{paramName}} !== undefined && {{paramName}} !== null) {
-                localVarHeaderParameter['{{baseName}}'] = String({{paramName}});
+                let mapped = (<any>"{{dataType}}" !== "string") ? JSON.stringify({{paramName}} || {}) : ({{paramName}} || "")
```

Comment:
@sruehl thanks for the PR. Can we use something like `{{#isString}} ... {{/isString}}` to check instead?

Reply:
to be honest I have no idea if its possible. What does this isString macro do?

Reply:
`isString` mustache tag is true if the data type is "string" 

(I noticed you compared datatype to "string" in the TS code)

Reply:
the code is the same as the one found [here](https://github.com/OpenAPITools/openapi-generator/blob/ccde3460805c1048067e241bc6a057ce92ce6a42/modules/openapi-generator/src/main/resources/typescript-axios/api.mustache#L228), so this class might need some more work

Reply:
btw. is there an example where to use it like if else?

Reply:
https://mustache.github.io/mustache.5.html#Inverted-Sections

Reply:
I changed it to the proposed macro